src/app/shared/issue-tables/issue-tables-paginated-data.ts

```diff
@@ -0,0 +1,14 @@
+import { MatPaginator } from '@angular/material';
+import { Issue } from '../../core/models/issue.model';
+
+export function setAndGetPaginatedData(paginator: MatPaginator, data: Issue[]): Issue[] {
+    paginator.length = data.length;
+    let startIndex = paginator.pageIndex * paginator.pageSize;
+    const result = data.splice(startIndex, paginator.pageSize);
+    if (result.length === 0) {
+        paginator.pageIndex -= 1;
+        startIndex = paginator.pageIndex * paginator.pageSize;
```

Comment:
We have a repeated code block:
```
startIndex = paginator.pageIndex * paginator.pageSize;
result = data.splice(startIndex, paginator.pageSize);
```

Let's move it into a separate method `getDataForPage`, which takes parameters `data`, `pageIndex` and `pageSize`